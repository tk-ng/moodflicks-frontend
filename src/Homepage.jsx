import { useState, useEffect } from "react";
import { Api } from "./api/api";
import { MoodActivities } from "./moodComponents/MoodActivities";

export const Homepage = () => {
	const [activities, setActivities] = useState([]);

	async function getActivities() {
		try {
			const result = await Api.getActivities();
			setActivities(result);
		} catch (err) {
			console.error(err);
		}
	}

	useEffect(() => {
		getActivities();
	}, []);

	return (
		<div className="Homepage p-5">
			<h1 className="Homepage-title pb-5">WELCOME TO MOODFLICKS!</h1>
			<h2>Discover movies that match your mood.</h2>
			{activities.length ? (
				<MoodActivities activities={activities} />
			) : (
				<p>Sorry, no results were found!</p>
			)}
			<div className="container mt-5">
				With MoodFlicks, you create personalized mood-based movie lists, explore
				curated vibes from other users, and get unique recommendations based on
				shared emotional themes—so you’re never stuck scrolling again. Whether
				you’re in the mood to laugh, cry, think, or chill—MoodFlicks gets it.
			</div>
			<div>Start building your vibe today.</div>
		</div>
	);
};
