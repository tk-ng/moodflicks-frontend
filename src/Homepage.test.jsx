import { Homepage } from "./Homepage";
import { render, screen } from "@testing-library/react";

vi.mock("../api/api", () => ({
	Api: {
		getActivities: vi.fn().mockResolvedValue([]),
	},
}));

describe("Renders the Homepage component", function () {
	it("mounts without crashing", async function () {
		const { container, asFragment } = render(<Homepage />);
		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("renders properly without activities", async function () {
		render(<Homepage />);
		expect(
			screen.getByText("Sorry, no results were found!")
		).toBeInTheDocument();
	});

	// With activities is tested in MoodActivities.test.jsx
});
