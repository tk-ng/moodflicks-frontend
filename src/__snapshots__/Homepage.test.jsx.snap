// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders the Homepage component > mounts without crashing 1`] = `
<DocumentFragment>
  <div
    class="Homepage p-5"
  >
    <h1
      class="Homepage-title pb-5"
    >
      WELCOME TO MOODFLICKS!
    </h1>
    <h2>
      Discover movies that match your mood.
    </h2>
    <p>
      Sorry, no results were found!
    </p>
    <div
      class="container mt-5"
    >
      With MoodFlicks, you create personalized mood-based movie lists, explore curated vibes from other users, and get unique recommendations based on shared emotional themes—so you’re never stuck scrolling again. Whether you’re in the mood to laugh, cry, think, or chill—MoodFlicks gets it.
    </div>
    <div>
      Start building your vibe today.
    </div>
  </div>
</DocumentFragment>
`;
